src/components/dashboard/components/Dashboard.js

```diff
@@ -1,24 +1,46 @@
 import React from 'react';
-import { Route } from 'react-router-dom';
+import { Route, Switch } from 'react-router-dom';
 
 import overview from '../../overview';
 import campaigns from '../../campaigns';
 import locations from '../../locations';
 import layouts from '../../layouts';
+import savedLocations from '../../savedLocations';
 
 const { Overview } = overview.components;
 const { Campaigns } = campaigns.components;
-const { Locations } = locations.components;
+const { Locations, LocationById } = locations.components;
+const { SavedLocations } = savedLocations.components;
 
 const { DashboardLayout } = layouts.components;
 const Dashboard = props => {
     const { path } = props.match;
     const getPathWay = pathRoute => `${path}/${pathRoute}`;
+    const getNestedPath = (pathRoute, child) => `${path}/${pathRoute}/${child}`;
     return (
         <DashboardLayout>
-            <Route path={path} exact strict component={Overview} />
-            <Route path={getPathWay('campaigns')} component={Campaigns} />
-            <Route path={getPathWay('locations')} component={Locations} />
+            <Switch>
+                <Route path={path} exact strict component={Overview} />
+                <Route path={getPathWay('campaigns')} exact strict component={Campaigns} />
+                <Route
+                    path={getPathWay('savedLocations')}
+                    exact
+                    strict
+                    component={SavedLocations}
+                />
+                <Route
+                    path={getPathWay('locations')}
+                    exact
+                    strict
+                    component={Locations}
+                />
+                <Route
+                    path={getNestedPath('locations', ':id')}
+                    exact
+                    strict
+                    component={LocationById}
+                />
```

Comment:
can you create these routes using array mapping. I am seeing a future where there will be more and more routes

Reply:
Alright. Will do that